src/main/java/nextstep/app/FilterConfig.java

```diff
@@ -0,0 +1,62 @@
+package nextstep.app;
+
+import nextstep.app.domain.Member;
+import nextstep.app.domain.MemberRepository;
+import nextstep.security.DefaultSecurityFilterChain;
+import nextstep.security.FilterChainProxy;
+import nextstep.security.SecurityFilterChain;
+import nextstep.security.filter.BasicAuthenticationFilter;
+import nextstep.security.filter.FormLoginAuthenticationFilter;
+import nextstep.security.UserDetails;
+import nextstep.security.UserDetailsService;
+import nextstep.security.util.matcher.AnyRequestMatcher;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.filter.DelegatingFilterProxy;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import java.util.List;
+
+@Configuration
+public class FilterConfig implements WebMvcConfigurer {
+
+    private final MemberRepository memberRepository;
+
+    public FilterConfig(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    @Bean
+    public DelegatingFilterProxy delegatingFilterProxy() {
+        return new DelegatingFilterProxy(filterChainProxy());
+    }
+
+    private FilterChainProxy filterChainProxy() {
+        return new FilterChainProxy(securityFilterChain());
+    }
+
+    private SecurityFilterChain securityFilterChain() {
+        return new DefaultSecurityFilterChain(AnyRequestMatcher.INSTANCE, List.of(
+                new BasicAuthenticationFilter(userDetailsService()),
+                new FormLoginAuthenticationFilter(userDetailsService())
+        ));
```

Comment:
```suggestion
        UserDetailsService userDetailsService = userDetailsService();
        return new DefaultSecurityFilterChain(AnyRequestMatcher.INSTANCE, List.of(
                new BasicAuthenticationFilter(userDetailsService),
                new FormLoginAuthenticationFilter(userDetailsService)
        ));
```
객체를 두번 생성하지 않고 변수로 빼서 한번만 사용할 수 있도록 변경할 수 있을 것 같아요 :)